### PR TITLE
Adding error handling (PROJQUAY-4151)

### PR DESCRIPTION
**Pull request by bcaton85**
**Read before reviewing**

Detailed error messages are not included in this PR and will be implemented in a future PR. This PR creates the structure required for sufficient error handling.

The size of this PR is due to implementing the same change across all components making API calls.

**Summary of changes**
- Catching and handling or error's in each component
- Addition of an [ErrorBoundary](https://reactjs.org/docs/error-boundaries.html) and various display errors.
- Refactoring of API calls
    - Removed catching and re-throwing of errors - errors will be caught and handled in the components that use them
    - Assertion of the expected HTTP status code
    - Calls to `getUser`("/api/v1/user/") are consolidated in the `UserState` selector
    - Restructuring of calls

**How to test**
Replace various responses in `test/fake-db/data` with 500 error codes and run `MOCK_API=true npm start`. Can also run the UI with no Quay backend.

### CODE REVIEW

File: src/atoms/UserState.ts [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
+  },
 });
 
 export const UserOrgs = selector<IOrganization[] | undefined>({
   key: 'userOrgs',
   get: async ({get}) => {
-    const user = get(UserState);
-    return user ? user.organizations : undefined;
+    const user: IUserResource = get(UserState);
+    // Check should never fail, if request was unsuccessful it throws
```
Comment (HammerMeetNail) [inline src/atoms/UserState.ts:31]: I feel like when I see this sort of comment, it also means if the error condition does happen it'll be harder to troubleshoot. Do you mind adding something to catch and log something to console at the minimum in the super unlikely event it ever catches?

File: src/components/modals/ConfirmationModal.tsx [hunk 2]
```diff
@@ -1,35 +1,43 @@
...
 
   const handleModalConfirm = async () => {
-    await changeVisibility().then(async () => {
-      const user = await getUser();
-      setUserState(user);
-      props.toggleModal();
-      props.selectAllRepos(false);
-    });
+    // This modal should never render if no items have been selected,
+    // that should be handled by the parent component. Leaving this check
```
Comment (HammerMeetNail) [inline src/components/modals/ConfirmationModal.tsx:34]: 👍

File: src/hooks/UseQuayConfig.ts [hunk 3]
```diff
@@ -10,9 +10,17 @@ export function useQuayConfig() {
 
   useEffect(() => {
     (async () => {
-      const config = await fetchQuayConfig();
-      setQuayConfig(config);
-      console.log(config);
+      // NOTE: Making the decision that loading the Quay configuration
+      // is not required. If the load fails the app will continue running.
+      // Components using this hook should check for null values. This
```
Comment (HammerMeetNail) [inline src/hooks/UseQuayConfig.ts:15]: I wonder if we should call this out in the readme? Config is already an odd call, maybe we should call it out in the future explaining why we even make the call.

File: src/routes/OrganizationsList/OrganizationsList.tsx [hunk 4]
```diff
@@ -41,6 +56,23 @@ export default function OrganizationsList() {
   const filter = useRecoilValue(filterOrgState);
   const [selectedOrganization, setSelectedOrganization] =
     useRecoilState(selectedOrgsState);
+  const userOrgs = useRecoilValue(UserOrgs);
+  const refreshUser = useRefreshUser();
+  useEffect(() => {
+    // Get latest organizations
+    refreshUser();
+  }, []);
+  // TODO: Using mock values here - remove in the future?
```
Comment (HammerMeetNail) [inline src/routes/OrganizationsList/OrganizationsList.tsx:65]: Yep, https://issues.redhat.com/browse/PROJQUAY-4082

File: src/routes/RepositoriesList/RepositoriesList.tsx [hunk 5]
```diff
@@ -164,32 +196,29 @@ export default function RepositoriesList() {
...
+        // API we should pass 'repos' directly into 'setRepositoryList'
+        const formattedRepos = repos.map((repo) => ({
+          namespace: repo.namespace,
+          name: repo.name,
+          is_public: repo.is_public,
+          last_modified: repo.last_modified,
+        }));
+        setRepositoryList(formattedRepos);
+      } catch (err) {
+        console.log('setting err');
```
Comment (HammerMeetNail) [inline src/routes/RepositoriesList/RepositoriesList.tsx:218]: Can you remove any of the `console.log` messages?

**Review by HammerMeetNail — APPROVED** [review]
Works locally, ran through a handful of error scenarios. Added a couple comments, nothing major. I appreciate all the comments.

File: src/hooks/UseQuayConfig.ts (hunk 3 above)
Comment (bcaton85) [reply src/hooks/UseQuayConfig.ts:15]: Yeah good point. I'm not sure if the readme is the right place for this info. Would that go into a design details section or something along those lines? Maybe something to discuss separately

File: src/routes/OrganizationsList/OrganizationsList.tsx (hunk 4 above)
Comment (bcaton85) [reply src/routes/OrganizationsList/OrganizationsList.tsx:65]: awesome

File: src/atoms/UserState.ts (hunk 1 above)
Comment (bcaton85) [reply src/atoms/UserState.ts:31]: True, added an error log.

File: src/routes/RepositoriesList/RepositoriesList.tsx (hunk 5 above)
Comment (bcaton85) [reply src/routes/RepositoriesList/RepositoriesList.tsx]: oops, this one snuck by. Removed.

**Pull request merged**